regolith/helpers/a_expensehelper.py

```diff
@@ -84,16 +87,28 @@ def construct_global_ctx(self):
         gtx["zip"] = zip
 
     def db_updater(self):
+        gtx = self.gtx
+
         rc = self.rc
-        if not rc.date:
-            now = dt.date.today()
+
+        if not rc.begin_date:
+            begin_date = dt.date.today() # string with date time format
+            begin_date = begin_date.strftime('%Y-%m-%d')
         else:
-            now = date_parser.parse(rc.date).date()
-        if not rc.due_date:
-            due_date = now + relativedelta(years=1)
+            begin_date = rc.begin_date() # need to raise exception for wrongly formatted user input
+        if not rc.end_date:
+            end_date = dt.date.today()
+            end_date = end_date.strftime('%Y-%m-%d')
         else:
-            due_date = date_parser.parse(rc.due_date).date()
-        key = f"{str(now.year)[2:]}{rc.lead[:2]}_{''.join(rc.name.casefold().split()).strip()}"
+            end_date = rc.end_date
+
+        # defaults to sbillinge
+        if rc.payee:
+            key = f"{str(begin_date.year)[2:]}{rc.payee[:2]}_{''.join(rc.name.casefold().split()).strip()}"
```

Comment:
pls check that this is the pattern for the key in expenses.  I am guessing not.

Reply:
I am actually not sure what defines the name of an expense. I am guessing a user specifies the name, but there are names like 2003sb_google, where does the 2003 come from?

Reply:
this is a moving target.  We never defined a heuristic for this, so you can come up with one, but my sense is that it is more informational if it has info about the expense rather than just the name of the payee.  maybe something like ``2006sb_<sthg>`` as an example.

Reply:
For cases like sstao, from what I see an expense name would be 20st_name, not 20ss_name.

Reply:
I'm not sure, but it could be a fun project for you to work on.  It crops up elsewhere in the project.  don't waste too much time on it though, it is not that big of a deal, especially in expenses.

btw, please make sure you go through and respond to all my comments.  That way I know that you saw them.

Reply:
Ok, for now I think it would be simpler to just ignore these exceptions.

Reply:
cool.  Do me a favor though and throw up that job as an issue.....

we probably want to have a function in tools that builds the name-based part of the ID and not have the code that does it in every helper.  there are two patterns, YYfl and YYMMfl and the function could do both.  Don't make the function, just make the issue and let me know when it is done.

Reply:
created new issue for the function.